### PR TITLE
feat(TDOPS-421): Pass data attributes to faceted search components

### PR DESCRIPTION
**Pull request by Gbacc**
**What is the problem this PR is trying to solve?**
Pass data attributes to faceted search components

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/faceted-search/src/components/Badges/BadgeDate/BadgeDate.component.js [hunk 1]
```diff
@@ -38,6 +38,7 @@ const BadgeDate = ({
 	readOnly,
 	removable,
 	displayType,
+	...rest
```
Comment (ybaskaran) [inline packages/faceted-search/src/components/Badges/BadgeDate/BadgeDate.component.js:41]: can we pass only the required props ? like dataAttributes ?

**Review by ybaskaran — APPROVED** [review]

File: packages/faceted-search/src/components/Badges/BadgeDate/BadgeDate.component.js (hunk 1 above)
Comment (Gbacc) [reply packages/faceted-search/src/components/Badges/BadgeDate/BadgeDate.component.js:41]: Well why not it's up to us i can do the change to restrict a bit

File: packages/faceted-search/src/components/BadgesGenerator/BadgesGenerator.component.js [hunk 2]
```diff
@@ -3,15 +3,21 @@ import PropTypes from 'prop-types';
...
+		const dataAttributesKeys = Object.keys(metadata).filter(objectKey =>
+			objectKey.startsWith('data-'),
+		);
+		const dataAttributes = pick(metadata, dataAttributesKeys);
 		if (BadgeComponent) {
 			acc.push(
 				<BadgeComponent
 					{...metadata}
 					{...properties}
+					dataAttributes={dataAttributes}
```
Comment (jmfrancois) [inline packages/faceted-search/src/components/BadgesGenerator/BadgesGenerator.component.js:20]: why not spread dataAttributes and use rest in others ?

**Review by jmfrancois — CHANGES_REQUESTED** [review]
I don't really like the `dataAttributes`
I would let a more global function getDataAttributeFrom(props) where you can read it.
So we can use spread
```{...getDataAttributeFrom(rest)}``` If you think there is other stuff in rest you want.

or [dataAttributes, others] = filterDataAttributesFrom(props) so you can keep the other.
Even better I think
WDYT ?

**Comment by Gbacc** [comment]
> I don't really like the `dataAttributes` I would let a more global function getDataAttributeFrom(props) where you can read it. So we can use spread `{...getDataAttributeFrom(rest)}` If you think there is other stuff in rest you want.
> 
> or [dataAttributes, others] = filterDataAttributesFrom(props) so you can keep the other. Even better I think WDYT ?

Ok let's say , in the badge generator i remove the filter on `data-`, you are right it's already spread with metadata.
In the upper badge component we just spread `{...rest}` like it's already done,  and in the badgeForm component we use some kind of function to filter data attributes only `{...getDataAttributeFrom(rest)}` on the Action ?

**Review by jmfrancois — APPROVED** [review]

**Pull request merged**